tools/clang/test/SemaHLSL/hlsl/types/invalid-hitobject-decls-struct.hlsl

```diff
@@ -0,0 +1,340 @@
+// expected-note@16{{'dx::HitObject' field declared here}}
+RWStructuredBuffer<TYPE> rw_struct_buf;
+// expected-error@-1{{object 'dx::HitObject' is not allowed in structured buffers}}
+// expected-note@16{{'dx::HitObject' field declared here}}
+ConstantBuffer<TYPE> const_buf;
+// expected-error@-1{{object 'dx::HitObject' is not allowed in ConstantBuffers or TextureBuffers}}
+// expected-note@16{{'dx::HitObject' field declared here}}
+TextureBuffer<TYPE> tex_buf;
+// expected-error@-1{{object 'dx::HitObject' is not allowed in ConstantBuffers or TextureBuffers}}
+// expected-note@16{{'dx::HitObject' field declared here}}
+
+ByteAddressBuffer bab;
+RWByteAddressBuffer rw_bab;
+
+[Shader("raygeneration")]
+void main()
+{
+   bab.Load<TYPE>(0);
+   // expected-error@-1{{Explicit template arguments on intrinsic Load must be a single numeric type}}
+   rw_bab.Load<TYPE>(0);
+   // expected-error@-1{{Explicit template arguments on intrinsic Load must be a single numeric type}}
+   TYPE val;
+   rw_bab.Store<TYPE>(0, val);
+   // expected-error@-1{{Explicit template arguments on intrinsic Store must be a single numeric type}}
+}
+
+[shader("pixel")]
+TYPE ps_main( 
+// expected-error@-1{{object 'dx::HitObject' is not allowed in entry function return type}}
+// expected-note@16{{'dx::HitObject' field declared here}}
+    TYPE vec : V) : SV_Target {
+    // expected-error@-1{{object 'dx::HitObject' is not allowed in entry function parameters}}
+    // expected-note@16{{'dx::HitObject' field declared here}}
+  return vec;
+}
+
+[shader("vertex")]
+TYPE vs_main(
+// expected-error@-1{{object 'dx::HitObject' is not allowed in entry function return type}}
+// expected-note@16{{'dx::HitObject' field declared here}}
+    TYPE parm : P) : SV_Target {
+    // expected-error@-1{{object 'dx::HitObject' is not allowed in entry function parameters}}
+    // expected-note@16{{'dx::HitObject' field declared here}}
+  parm.f = 0;
+  return parm;
+}
+
+
+[shader("geometry")]
+[maxvertexcount(3)]
+void gs_point(
+    line TYPE e,
+    // expected-error@-1{{object 'dx::HitObject' is not allowed in entry function parameters}}
+    // expected-note@16{{'dx::HitObject' field declared here}}
+    inout PointStream<TYPE> OutputStream0)
+    // expected-error@-1{{object 'dx::HitObject' is not allowed in geometry streams}}
+    // expected-note@16{{'dx::HitObject' field declared here}}
+{}
+
+[shader("geometry")]
+[maxvertexcount(12)]
+void gs_line( 
+    line TYPE a,
+    // expected-error@-1{{object 'dx::HitObject' is not allowed in entry function parameters}}
+    // expected-note@16{{'dx::HitObject' field declared here}}
+    inout LineStream<TYPE> OutputStream0)
+    // expected-error@-1{{object 'dx::HitObject' is not allowed in geometry streams}}
+    // expected-note@16{{'dx::HitObject' field declared here}}
+{}
+
+
+[shader("geometry")]
+[maxvertexcount(12)]
+void gs_line(
+    line TYPE a,
```

Comment:
This is a flaw copied from my own work in incomplete-type.hlsl and invalid-longvec-decls.hlsl, but this should have been for triangles
```suggestion
void gs_tri(
    triangle TYPE a,
```